src/shogun/lib/SGMatrix.h

```diff
@@ -41,7 +41,15 @@ template<class T> class SGMatrix : public SGReferencedData
 	public:
 		typedef RandomIterator<T> iterator;
 		typedef ConstRandomIterator<T> const_iterator;
+#ifndef SWIG	
```

Comment:
@gf712 do we have to put all this code into SGMatrix.h? It bloats the file quite a bit? I'd prefer to have it somewhere else, where other iterators sit. But you or @vigsterkr might see this differently?

Reply:
Not a pro opinion but, I did try to have a go at pushing all this code to our ```<shogun/utils/iterators.h>```. This, however, creates a circular dependency issue because our iterator class  SGMatrix as an internal member and SGMatrix #includes iterator file.

Reply:
Yes, I feared that. Maybe the others have an idea

Reply:
I mean it is doable, not sure if it will help. So you could have the declaration in iterator.h with forward declaration in SGMatrix and then implement the iterator in SGMatrix.cpp. This should work.